tests/backup/backup_helper.go

```diff
@@ -946,25 +946,28 @@ func CreateRestoreWithoutCheck(restoreName string, backupName string,
 }
 
 // CreateRestoreWithValidation creates restore, waits and checks for success and validates the backup
-func CreateRestoreWithValidation(ctx context.Context, restoreName, backupName string, namespaceMapping, storageClassMapping map[string]string, clusterName string, orgID string, scheduledAppContexts []*scheduler.Context) (err error) {
-	err = CreateRestore(restoreName, backupName, namespaceMapping, clusterName, orgID, ctx, storageClassMapping)
+func CreateRestoreWithValidation(ctx context.Context, restoreName, backupName string, namespaceMapping, storageClassMapping map[string]string, clusterName string, orgID string, scheduledAppContexts []*scheduler.Context) error {
+	err := CreateRestore(restoreName, backupName, namespaceMapping, clusterName, orgID, ctx, storageClassMapping)
 	if err != nil {
-		return
+		return err
 	}
 	originalClusterConfigPath := CurrentClusterConfigPath
 	if clusterConfigPath, ok := ClusterConfigPathMap[clusterName]; !ok {
 		err = fmt.Errorf("switching cluster context: couldn't find clusterConfigPath for cluster [%s]", clusterName)
-		return
+		return err
 	} else {
 		log.InfoD("Switching cluster context to cluster [%s]", clusterName)
 		err = SetClusterContext(clusterConfigPath)
 		if err != nil {
-			return
+			return err
 		}
 	}
 	defer func() {
 		log.InfoD("Switching cluster context back to cluster path [%s]", originalClusterConfigPath)
-		err = SetClusterContext(originalClusterConfigPath)
+		err := SetClusterContext(originalClusterConfigPath)
+		if err != nil {
+			log.Errorf("Failed switching cluster context back to cluster path [%s].\nError - %s", originalClusterConfigPath, err.Error())
```

Comment:
How about log.FailOnError?

Reply:
I can do `log.FailOnError`
But not sure how will it work if I just append to the error that is being returned by the function.

Reply:
The test case will fail right if err != nil?

Reply:
Added `log.FailOnError`